### ISSUE TITLE
Portus crashes when guessing LDAP email

### ISSUE THREAD

**Issue by alexanderteves** [SUSE/Portus#1832]
### Description

Portus crashes when guessing LDAP email while working fine without it.

### Steps to reproduce

I set up Portus to use LDAP which works fine without guessing the email. But when enabling guessing with the actual attribute, Portus crashes with:

```
NoMethodError (undefined method `size' for nil:NilClass):
  lib/portus/ldap.rb:235:in `guess_email'
  lib/portus/ldap.rb:213:in `find_or_create_user!'
  lib/portus/ldap.rb:195:in `portus_login!'
  lib/portus/ldap.rb:43:in `authenticate!'
  app/controllers/application_controller.rb:56:in `force_update_profile!'
  app/middleware/catch_json_parse_errors.rb:11:in `call'
```

At least I would expect it to fall back to no email setup and redirecting the user to the profile page to enter it manually as stated in the docs.

### Deployment information

**Deployment method**:

Bare metal with official Docker image v2.3

**Configuration**:

```yml
ldap:
  enabled: true

  hostname: "ldap.example.com"
  port: 389
  method: "start_tls"

  encryption:
    method: "plain"
  
base: "dc=example,dc=com"

  filter: "(&(objectClass=person)(|(employeeType=temporary)(employeeType=permanent)(employeeType=technical)(employeeType=external)))"

  uid: "uid"

  authentication:
    enabled: true
    bind_dn: "cn=readonly,ou=admin,dc=example,dc=com"
    password: "secret123"

  guess_email:
    enabled: true
    attr: "mail"
```

**Portus version**: v2.3

**Comment by mssola** [comment]
There are two things to note here:

1. This is because a `search` fails, and it will return `nil` apparently if it failed as specified [here](https://www.rubydoc.info/gems/ruby-net-ldap/Net/LDAP#search-instance_method). So, it's definitely a bug, because we don't protected ourselves from that when guessing the email...
2. Notice that your configuration is flawed: the `base` attribute should be at the same indentation level as `hostname` or `port`. That will definitely cause some problems (and possibly it will lead to this bug :wink:).

Thanks for the report ! :+1:

**Comment by alexanderteves** [comment]
@mssola Thanks for the heads up. Concerning 2): My mistake while pasting it here. I checked with our production *.yml and the indention is correct there, still fails. Sorry for that!

**Comment by alexanderteves** [comment]
While I have seen your fix, it still bothers me that it crashes in my scenario. When the LDAP authentication works absolutely fine, I wonder how each search should return `nil` when guessing the email? Every record has an attribute `mail` which holds the actual address. Debug mode sadly does not yield anything useful. Or maybe I miss some basic knowledge ;)

**Comment by alexanderteves** [comment]
I debugged the `@ldap.search` in L234 with `@ldap.get_operation_result` and the message is `unauthenticated bind (DN with no password) disallowed`. True, I don't want that. But is that not what the credentials from `authentcation:` should be used for?

**Comment by mssola** [comment]
I've also adapted the fix into `2.3`: 93df51cce0da1d978c4c4850b188b6e399b18a59. It should be available on the Docker image rather shortly.
